### PR TITLE
feat: force password change

### PR DESCRIPTION
**Pull request by NaMax66**
Description:
https://virtocommerce.atlassian.net/browse/ST-4542

--

QA-test:

Demo-test:

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.35.0-pr-635-7d2f-7d2f9d49.zip

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -280,6 +280,9 @@ module.exports = {
         "screen-60": "60vh",
         "screen-75": "75vh",
       },
-    },
+      zIndex: {
```
Comment (goldenmaya) [inline tailwind.config.js:283]: ?

File: client-app/ui-kit/components/organisms/popup/vc-popup.vue [hunk 2]
```diff
@@ -14,7 +14,7 @@
       </TransitionChild>
 
       <div
-        class="fixed inset-0 z-50"
+        class="fixed inset-0 z-5000"
```
Comment (goldenmaya) [inline client-app/ui-kit/components/organisms/popup/vc-popup.vue:17]: ?

File: client-app/pages/cart.vue [hunk 3]
```diff
@@ -341,6 +341,8 @@ invoke(async () => {
   /**
    * Send a Google Analytics shopping cart view event.
    */
-  ga.viewCart(cart.value!);
+  if (cart.value) {
```
[the remark this answers is not in the document]
Comment (NaMax66) [reply client-app/pages/cart.vue:344]: prevent uncaught error

File: client-app/shared/broadcast/composables/useBroadcast.ts [hunk 4]
```diff
@@ -3,6 +3,11 @@ import { noop, tryOnScopeDispose, useEventListener } from "@vueuse/core";
 import { Logger } from "@/core/utilities";
 import type { InjectionEvent } from "../types";
 
+export enum TabsType {
```
Comment (goldenmaya) [inline client-app/shared/broadcast/composables/useBroadcast.ts:6]: We have another PR with these changes, so these changes need to be rolled back.

File: tailwind.config.js [hunk 5]
```diff
@@ -279,7 +279,7 @@ module.exports = {
       maxHeight: {
         "screen-60": "60vh",
         "screen-75": "75vh",
-      },
-    },
+      }
```
Comment (goldenmaya) [inline tailwind.config.js:282]: rollback changes

File: client-app/pages/account/profile.vue [hunk 6]
```diff
@@ -191,43 +187,64 @@ const { value: oldPassword } = useField<string>("oldPassword");
...
+      errorDescription = t("shared.account.change_password_form.errors.wrong_current_pass");
+    } else {
+      errorDescription = getIdentityErrorTranslation(error);
+    }
+
+    if (errorDescription) {
+      commonErrors.value.push(errorDescription);
+    }
+  });
+};
```
Comment (asvishnyakov) [inline client-app/pages/account/profile.vue:204]: This duplicated code should probably be moved into composable

File: client-app/broadcast.ts [hunk 7]
```diff
@@ -32,4 +37,14 @@ export function setupBroadcastGlobalListeners() {
       await fetchShortCart();
     }
   });
+  on(unauthorizedErrorEvent, async () => {
+    await fetchUser();
+    if (!isPasswordNeedToBeChanged.value) {
+      const { hash, pathname, search } = location;
+
+      if (pathname !== "/sign-in") {
+        location.href = `/sign-in?returnUrl=${pathname + search + hash}`;
```
Comment (asvishnyakov) [inline client-app/broadcast.ts:46]: Shouldn't we use `router` here and below where you use `location`?

File: client-app/shared/account/components/change-password-form.vue [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+import { usePasswordRequirements, useUser } from "../composables";
+import PasswordTips from "./password-tips.vue";
+
+interface IEmits {
+  (event: "succeeded"): void;
+}
+
+const emit = defineEmits<IEmits>();
+
+const MAX_PASS_LENGTH = 64;
```
Comment (asvishnyakov) [inline client-app/shared/account/components/change-password-form.vue:69]: Why?

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

**Review by asvishnyakov — COMMENTED** [review]
For history: I would expect to reuse/refactor this: https://github.com/VirtoCommerce/vc-theme-b2b-vue/blob/dev/client-app/shared/account/components/reset-password-form.vue

File: client-app/pages/account/profile.vue (hunk 6 above)
Comment (NaMax66) [reply client-app/pages/account/profile.vue:204]: Great Idea! I'll make a task

File: client-app/broadcast.ts (hunk 7 above)
Comment (NaMax66) [reply client-app/broadcast.ts:46]: The App should be reloaded to clear state

File: client-app/pages/account/profile.vue (hunk 6 above)
Comment (NaMax66) [reply client-app/pages/account/profile.vue]: Keep standrad

**Review by asvishnyakov — APPROVED** [review]

**Pull request merged**